samples/mlhelib_tutorial/tut_4_integers.cpp

```diff
@@ -0,0 +1,224 @@
+/*
+* MIT License
+*
+* Copyright (c) 2020 International Business Machines
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in all
+* copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+* SOFTWARE.
+*/
+
+#include <cassert>
+
+#include "helayers/hebase/helib/HelibBgvContext.h"
+#include "helayers/hebase/hebase.h"
+
+// In this tutorial we'll use a different scheme.
+// So far we used CKKS that works with complex numbers,
+// Here we'll configure HElib's BGV as the underlyin scheme,
+// which works with integer numbers.
+// Hence it has some limitations, and some
+// additional operators.
+
+using namespace std;
+using namespace helayers;
+
+void tut_4_run(HeContext& he);
+
+void tut_4_integers()
+{
+  // First, we initialize the library
+  // As before, this performs all the required setup.
+  // NOTE: this also generates a public-private key pair,
+  // so it's ready now for use.
+  int p = 4999;  // Plaintext prime modulus
```

Comment:
maybe some better names vars would be helpful, something more descriptive than 1 letter?